go/common/crypto/signature/signers/ledger/ledger_signer.go

```diff
@@ -0,0 +1,43 @@
+package ledger
+
+import (
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	ledger "github.com/zondax/ledger-oasis-go"
+)
+
+var (
+	defaultDerivationPath []uint32 = []uint32{44, 118, 0, 0, 0}
+)
+
+type Signer struct {
+	app *ledger.LedgerOasis
+}
+
+// NewSigner will attempt to initialize a Ledger Signer by running
+// FindLedger. This will connect to the first Ledger device found.
+func NewSigner() (signature.Signer, error) {
+	app, err := ledger.FindLedgerOasisApp()
+	if err != nil {
+		return nil, err
+	}
+	return &Signer{app}, nil
+}
+
+func (s *Signer) Public() signature.PublicKey {
+	var pubKey signature.PublicKey
+	retrieved, _ := s.app.GetPublicKeyEd25519(defaultDerivationPath)
```

Comment:
The error is thrown out here. I don't think we want this, but the process of changing the `Signer` interface is pretty tedious, so I won't do it unless we can't think of a better solution.